plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt

```diff
@@ -247,6 +276,11 @@ open class KtlintPlugin : Plugin<Project> {
         return {
             val ktLintConfig = createConfiguration(project, extension)
 
+            project.components.withType(KotlinNativeComponent::class.java).ifNotEmpty {
```

Comment:
Remind me the difference between `components` and `extensions`?

Reply:
`Component` is configuration for some compile/output artifact, see https://docs.gradle.org/current/javadoc/org/gradle/api/component/SoftwareComponent.html